app/models/work.rb

```diff
@@ -0,0 +1,53 @@
+class Work < ApplicationRecord
+  has_one :vote
+  has_many :users, :through => :vote
+  after_create :create_vote
+  validates :title, presence: true, uniqueness: true
+  validates :creator, presence: true
+  validates :year, presence: true
+  validates :description, presence: true
+ 
+  def create_vote
+    vote = Vote.create(work_id: self.id)
+  end  
+
+  def self.find_top_amount(type)
+    works = Work.where(category: type)
+    works = works.sort_by { |work| work.vote.users.length }
```

Comment:
I just want to make the point that it's more efficient to use the `.order` method to sort the list of works and let Postgres do the sorting.  It's faster than retrieving all the works and then sorting them.